### PR TITLE
WIP: Large Clusters V2

### PR DESCRIPTION
**Pull request by khenidak**
Implements proposal outlined here: https://github.com/Azure/aks-engine/issues/443

### CODE REVIEW

File: examples/largeclusters/v2/create-etcd.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+err(){
+	local msg="$1"
+	echo -e "${Red}ERR:${msg}"
+	exit 1
+}
+
+#creates random string (starts with letter)
+get_random(){
+	local length="$1"
+	if [[ -z "${length}" ]] || [[ "0" -eq "$lenght" ]]; then
```
Comment (tariq1890) [inline examples/largeclusters/v2/create-etcd.sh]:
```suggestion
	if [[ -z "${length}" ]] || [[ "0" -eq "$lengtht" ]]; then
```

File: examples/largeclusters/v2/create-etcd.sh [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+# Expected input
+#SUB_ID							subscription-id									optional(default what az is configured to use)
+#RG									resource_group 									mandatory
+#VNET_NAME					vnet name												mandatory
+#SUBNET_NAME				subnet name											mandatory
+#BASE_CLUSTER_NAME	basename for cluster resourcs		optional(default random)
+#CLUSTER_SIZE				etcd cluster size								optional(default 3)
+#VM_SIZE						etcd vm size										optional(default: Standard_DS2_v2)
+#OUTPUT_DIR					locatio of generated certs			optional(default /tmp/${BASE_CLUSTER_NAME}
```
Comment (tariq1890) [inline examples/largeclusters/v2/create-etcd.sh]:
```suggestion
#OUTPUT_DIR					location of generated certs			optional(default /tmp/${BASE_CLUSTER_NAME}
```

File: examples/largeclusters/v2/create-etcd.sh [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+set -eo pipefail
+
+CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Expected input
+#SUB_ID							subscription-id									optional(default what az is configured to use)
+#RG									resource_group 									mandatory
+#VNET_NAME					vnet name												mandatory
+#SUBNET_NAME				subnet name											mandatory
+#BASE_CLUSTER_NAME	basename for cluster resourcs		optional(default random)
```
Comment (tariq1890) [inline examples/largeclusters/v2/create-etcd.sh]:
```suggestion
#BASE_CLUSTER_NAME	basename for cluster resources		optional(default random)
```

File: examples/largeclusters/v2/readme.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+##Create Resource Group
+
+```
+az group create --name <<resource_group_name>> --location <<azure_location>>
+```
+
+
+##Create vnet and etcd subnet
+
+> You only need to do that if you are creating the cluster in a non exitent vnet
```
Comment (CecileRobertMichon) [inline examples/largeclusters/v2/readme.md]: non-existent?

File: examples/largeclusters/v2/readme.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#subnet address prefix: 10.0.1.0/24 (256 - 4)
+az network vnet create --resource-group <<resource_group_name>> \
+                       --name <<vnet_name>> \
+                       --address-prefix <<vnet address space>> \
+                       --subnet-name <<etcd_subnet_name>>  \
+                       --subnet-prefix <<etcd_subnet_address_space>>
+```
+
+
+> Depending on your cluster size you may need to create 2 or more etcd cluster. For exampl an etcd cluster for `events` another for everything else.
```
Comment (CecileRobertMichon) [inline examples/largeclusters/v2/readme.md]: Depending on your cluster size, you may need to create 2 or more etcd clusters. For example, an etcd cluster for `events` and another for everything else.
```suggestion
> Depending on your cluster size, you may need to create 2 or more etcd clusters. For example, an etcd cluster for `events` and another for everything else.
```

**Comment by khenidak** [comment]
I found a way to run 1000-1500 node using our existing setup without the complexity of external etcd. I will be updating this WIP PR accordingly.

**Pull request closed without merging**